stdlib/multiprocessing/util.pyi

```diff
@@ -0,0 +1,122 @@
+import threading
+from _typeshed import Self, StrOrBytesPath
+from collections.abc import Callable, Iterable, Sequence
+from itertools import count
+from logging import Logger
+from typing import Any
+from weakref import WeakValueDictionary
+
+from . import process
+
+__all__ = [
+    "sub_debug",
+    "debug",
+    "info",
+    "sub_warning",
+    "get_logger",
+    "log_to_stderr",
+    "get_temp_dir",
+    "register_after_fork",
+    "is_exiting",
+    "Finalize",
+    "ForkAwareThreadLock",
+    "ForkAwareLocal",
+    "close_all_fds_except",
+    "SUBDEBUG",
+    "SUBWARNING",
+]
+
+NOTSET: int = ...
+SUBDEBUG: int = ...
+DEBUG: int = ...
+INFO: int = ...
+SUBWARNING: int = ...
+
+LOGGER_NAME: str = ...
+DEFAULT_LOGGING_FORMAT: str = ...
+
+_logger: Logger | None = ...
+_log_to_stderr: bool = ...
+
+def sub_debug(msg: object, *args: object) -> None: ...
+def debug(msg: object, *args: object) -> None: ...
+def info(msg: object, *args: object) -> None: ...
+def sub_warning(msg: object, *args: object) -> None: ...
+def get_logger() -> Logger: ...
+def log_to_stderr(level: int | None = ...) -> Logger: ...
+def _platform_supports_abstract_sockets() -> bool: ...
+def is_abstract_socket_namespace(address: str | bytes | None) -> bool: ...
+
+abstract_sockets_supported: bool = ...
+
+def _remove_temp_dir(rmtree: Callable[[StrOrBytesPath], None], tempdir: StrOrBytesPath) -> None: ...
+def get_temp_dir() -> str: ...
+
+_afterfork_registry: WeakValueDictionary[tuple[int, int, Callable[[Any], Any]], object]
+_afterfork_counter: count[int]
+
+def _run_after_forkers() -> None: ...
+def register_after_fork(obj: object, func: Callable[[Any], Any]) -> None: ...
+
+_finalizer_registry: dict[tuple[int | None, int], Finalize] = ...
+_finalizer_counter: count[int] = ...
+
+class Finalize:
+    _args: Any
+    _kwargs: dict[Any, Any]
+    _callback: Callable[[Any], Any]
+    _key: tuple[int | None, int]
+    _pid: int
+
+    def __init__(
+        self,
+        obj: object | None,
```

Comment:
Are you sure a user is _meant_ to be passing in literally any object? Or would `_typeshed.Incomplete` be better here? :)

Reply:
Putting in as `Incomplete | None` as the signature for `obj: object | None` was inferred based on the requirements of  `weakref.ref` from how it was used and checks performed on it explicitly include `is not None`

Reply:
This may make sense to overload, as `obj` and `exitpriority` go hand in hand in the implementation here:

https://github.com/python/cpython/blob/f40bc7fa49f8d137f9b38f7a14569e9a9bdbed07/Lib/multiprocessing/util.py#L192-L195

So potentially having one where `obj` None and `exitpriority` is required and one where it's `Incomplete`

Reply:
> This may make sense to overload

I think it probably isn't worth the complexity, given that this entire module is undocumented :)

Reply:
That's reasonable, I'll remove the todo comment next to that